src/components/Organisms/MarketingPreferencesDS/_MarketingPrefsConfig.js

```diff
@@ -108,7 +114,7 @@ const buildValidationSchema = overrideOptions => {
     mp_postcode: yup.string().when('mp_permissionPost', {
       is: val => (!(mpValidationOptions.mp_permissionPost.disableOption)
       && mpValidationOptions.mp_permissionPost[val]),
-      then: schema => schema.required('Please enter your postcode').matches(/^[a-zA-Z]{1,2}\d[a-zA-Z\d]?\s*\d[a-zA-Z]{2}$/, 'Please enter a valid postcode')
+      then: schema => schema.required('Please enter your postcode').test('postcode-valid', 'Please enter a valid postcode', validatePostcode)
```

Comment:
Thanks for this, makes a lot of sense to update both lookup components!

With Yup this logic will need to be done in two steps; the first is a `transform` which should apply `fixPostcode` to string values (and return anything else untouched), the second is the validity test which should be simply `postcode.isValid`. If we have only the test, Yup will accept the "fixed" value but not return it to us, and we'll get the original un-fixed value passed into the submission payload.

Something else to double-check, that we are actually using the return value from `mpValidationSchema.validate(...)`.